scripts/add_brownfield.py

```diff
@@ -0,0 +1,270 @@
+                n.links.loc[h2_retrofitted_fixed_i, "p_nom"]
+                .rename(lambda x: x.split("-2")[0].replace(fr, to))
+                .groupby(level=0)
+                .sum()
+            )
+            remaining_capacity = (
+                pipe_capacity
+                - CH4_per_H2
+                * already_retrofitted.reindex(index=pipe_capacity.index).fillna(0)
+            )
+            n.links.loc[gas_pipes_i, "p_nom"] = remaining_capacity
+        else:
+            new_pipes = n.links.carrier.isin(pipe_carrier) & (
+                n.links.build_year == year
+            )
+            n.links.loc[new_pipes, "p_nom"] = 0.0
+            n.links.loc[new_pipes, "p_nom_min"] = 0.0
+
+
+def disable_grid_expansion_if_limit_hit(n):
+    """
+    Check if transmission expansion limit is already reached; then turn off.
+
+    In particular, this function checks if the total transmission
+    capital cost or volume implied by s_nom_min and p_nom_min are
+    numerically close to the respective global limit set in
+    n.global_constraints. If so, the nominal capacities are set to the
+    minimum and extendable is turned off; the corresponding global
+    constraint is then dropped.
+    """
+    cols = {"cost": "capital_cost", "volume": "length"}
+    for limit_type in ["cost", "volume"]:
+        glcs = n.global_constraints.query(
+            f"type == 'transmission_expansion_{limit_type}_limit'"
+        )
+
+        for name, glc in glcs.iterrows():
+            total_expansion = (
+                (
+                    n.lines.query("s_nom_extendable")
+                    .eval(f"s_nom_min * {cols[limit_type]}")
+                    .sum()
+                )
+                + (
+                    n.links.query("carrier == 'DC' and p_nom_extendable")
+                    .eval(f"p_nom_min * {cols[limit_type]}")
+                    .sum()
+                )
+            ).sum()
+
+            # Allow small numerical differences
+            if np.abs(glc.constant - total_expansion) / glc.constant < 1e-6:
+                logger.info(
+                    f"Transmission expansion {limit_type} is already reached, disabling expansion and limit"
+                )
+                extendable_acs = n.lines.query("s_nom_extendable").index
+                n.lines.loc[extendable_acs, "s_nom_extendable"] = False
+                n.lines.loc[extendable_acs, "s_nom"] = n.lines.loc[
+                    extendable_acs, "s_nom_min"
+                ]
+
+                extendable_dcs = n.links.query(
+                    "carrier == 'DC' and p_nom_extendable"
+                ).index
+                n.links.loc[extendable_dcs, "p_nom_extendable"] = False
+                n.links.loc[extendable_dcs, "p_nom"] = n.links.loc[
+                    extendable_dcs, "p_nom_min"
+                ]
+
+                n.global_constraints.drop(name, inplace=True)
+
+
+# def adjust_renewable_profiles(n, input_profiles, params, year):
```

Comment:
Why is this not required?

Reply:
This optionally updates the input renewable time series for the components across years.
When I tested, it lead to problems and I considered of secondary priority but worth keeping to remind of future updates.
What do you think?

Reply:
Ah I see, interesting point. How is it handled currently, if I do a overnight scenario for 2030 and an overnight for 2050, the profiles are the same, right? If that's the case, we don't have to implement this feature here now but leave it for future developments.

Reply:
Exactly, that's the point. That's why it is also kept commented.
I resolve this comment, feel free to reopen it if appropriate :)